rust/ql/lib/codeql/rust/controlflow/internal/ControlFlowGraphImpl.qll

```diff
@@ -141,10 +141,10 @@ class LogicalAndTree extends PostOrderTree, LogicalAndExpr {
 
 class BlockExprTree extends StandardPostOrderTree, BlockExpr {
   override AstNode getChildNode(int i) {
-    result = super.getStmtList().getStatement(i)
+    result = super.getStmtList().getStatement(i) and
+    not result instanceof Item
     or
-    not exists(super.getStmtList().getStatement(i)) and
-    (exists(super.getStmtList().getStatement(i - 1)) or i = 0) and
+    i = super.getStmtList().getNumberOfStatements() and
```

Comment:
Yes, the children are later `rank`ed to ensure consecutive numbers. In any case the previous definition would not ensure gap-free results either. I think it could actually behave very curiously in the presence of several gaps.

I changed to `getNumberOfStatements` because that is much easier to understand, not to fix any bugs.

Reply:
Looks good then.  I agree `getNumberOfStatements` is cleaner.